### PR TITLE
Column breakpoints UI

### PR DESCRIPTION
**Pull request by audreyfang**
### Summary of Changes

Very rough start to the column breakpoint changes @darkwing and I discussed earlier. When alt/option is pressed column breakpoints are shown only for the function where the cursor is currently located. 

This is still quite buggy but works in some cases and I will also be further looking into changing the >> to actual icons.

### Screenshots/Videos

column-breakpoint

### CODE REVIEW

File: src/components/Editor/CallSites.js [hunk 1]
```diff
@@ -144,15 +145,34 @@ class CallSites extends Component {
...
-    const { editor, callSites, selectedSource } = this.props;
+    const { editor, callSites, selectedSource, symbols } = this.props;
     const { showCallSites } = this.state;
+
     let sites;
     if (!callSites) {
       return null;
     }
 
+    const { line } = editor.codeMirror.getCursor();
```
Comment (AnshulMalik) [inline src/components/Editor/CallSites.js]: I think `getCursor` gives us the position of the blinking cursor, and not mouse position, I am not sure though.

**Comment by darkwing** [comment]
Interesting that you've gone the CSS route instead of adding an element as a marker -- I look forward to seeing where this goes.  The important test for it is ensuring that clicking in that area of the marker does in fact activate/deactivate the column breakpoint -- CSS can be weird like that sometimes.

File: src/components/Editor/CallSite.css [hunk 2]
```diff
@@ -4,19 +4,20 @@
...
-  position: absolute;
+  content: ">>";
   width: 100%;
   height: calc(100% - 2px);
   border-bottom: 2px solid #aed3ef;
 }
 
 .call-site-bp {
   position: relative;
+  border-bottom: 2px solid red;
```
Comment (darkwing) [inline src/components/Editor/CallSite.css]: I'm guessing this is for debugger?

File: src/components/Editor/CallSite.css [hunk 3]
```diff
@@ -25,8 +26,7 @@
 }
 
 .call-site-bp::before {
-  content: "";
-  position: absolute;
+  content: ">>";
```
Comment (darkwing) [inline src/components/Editor/CallSite.css]: You should be able to use the same breakpoint svg image as the sidebar.

File: src/components/Editor/CallSites.js [hunk 4]
```diff
@@ -144,15 +145,34 @@ class CallSites extends Component {
...
+
+    let callSitesFiltered;
+    if (closestFunc !== null) {
+      callSitesFiltered = callSites.filter(
+        ({ location }) =>
+          location.start.line >= closestFunc.location.start.line &&
+          location.start.line <= closestFunc.location.end.line
+      );
+    } else {
+      callSitesFiltered = callSites;
```
Comment (darkwing) [inline src/components/Editor/CallSites.js]: You don't need the `else`; just initialize `callSitesFiltered` to `callSites`:

```js
let callSitesFiltered = callSites;
if ( ....
```

File: src/components/Editor/CallSites.js (hunk 1 above)
Comment (audreyfang) [reply src/components/Editor/CallSites.js]: It does give the position of the blinking cursor, which was my intention for this first iteration. Do you happen to know of any way to get the mouse hover position?

File: src/components/Editor/CallSite.css (hunk 2 above)
Comment (audreyfang) [reply src/components/Editor/CallSite.css]: Not completely sure I understand the question, but the red border is the current way of indicating that a breakpoint is selected. I added it here so that both the icon and text continue to be underlined.

Comment (darkwing) [reply src/components/Editor/CallSite.css]: Ahh, I see.  We probably want to find a different color, but that's a simple change so don't hang up on it.

File: src/components/Editor/CallSites.js (hunk 1 above)
Comment (AnshulMalik) [reply src/components/Editor/CallSites.js]: I did read the docs but couldn't find the mouse position, I think we need to come up with our own way to capture mouse position.

Comment (darkwing) [reply src/components/Editor/CallSites.js]: Let's table the mouse position experiment for now.  Let's just focus on clicking the icon adding the column breakpoint, and then having different states like enabled, disabled, off, as well as dark mode 💯

**Comment by darkwing** [comment]
Update as to what's left here @audreyfang ?

**Comment by audreyfang** [comment]
@darkwing I need to make a few minor fixes to the ui colours, look at your suggestions from last week and the failing tests. I should have another commit up later tonight.

**Comment by audreyfang** [comment]
@darkwing Do you have a specific example for when clicking on a breakpoint doesn't remove it? It works for me in most cases, although I do seem to have some issues with one particular function in the file I'm using.

File: assets/module-manifest.json [hunk 5]
```diff
@@ -20574,7 +20574,8 @@
       "../../parse5/lib/serializer/serializer_stream.js": 1752,
```
Comment (darkwing) [inline assets/module-manifest.json]: This doesn't belong in this PR, we'll remove it before merging.

File: assets/panel/prefs.js [hunk 6]
```diff
@@ -50,7 +50,7 @@ pref("devtools.debugger.skip-pausing", false);
 pref("devtools.debugger.features.wasm", true);
 pref("devtools.debugger.features.shortcuts", true);
 pref("devtools.debugger.features.root", true);
-pref("devtools.debugger.features.column-breakpoints", false);
+pref("devtools.debugger.features.column-breakpoints", true);
```
Comment (darkwing) [inline assets/panel/prefs.js]: 👍

Comment (darkwing) [reply assets/panel/prefs.js]: Actually, as things stand, we'll want to change this back to `false` before merging.  I just pressed `option` in a minified file and my browser hung, so we'll need to measure performance implications here.  Keep it true for now, if only for the sake of everyone testing and tests running.

**Comment by darkwing** [comment]
OMG I love how this looks so far.  Awesome awesome awesome!

I'm seeing a really weird issue and I'm wondering if @loganfsmyth or @AnshulMalik could help.  I prettified the following URL during testing:

https://davidwalsh.name/wp-content/themes/punky/js/main.js

I press option and see a bunch of lovely column breakpoint locations (yay!).  When I click one, however, the following issues occur:

1.  A breakpoint is added at the *next line*
2.  The column's arrow very briefly stays dark (yay!) but then goes back to inactive state (the root cause is probably the same as 1)
3.  I can't add multiple column breakpoints for the same line; instead the existing breakpoint's location is changed.

@loganfsmyth Can you look into the breakpoint location issue?  It's acting a bit odd.

@audreyfang In the mean time, could you explore adding mochitest for this functionality?  i.e. clicking a column breakpoint on and off does what's expected?  I expect you'll probably have questions here which we can discuss in Slack or 1:1 video. 

Great work so far!

**Comment by AnshulMalik** [comment]
It's the issue with source-map, either the source map generated from prettifying is not right(I think this is the case). Or `source-map` package is producing wrong results for generatedLocation.

**Comment by audreyfang** [comment]
@darkwing Will the column breakpoints remain enabled by default? If so, I need to update the jest tests to reflect that. If not, then this should be fine to merge.

**Comment by darkwing** [comment]
@audreyfang I think you'll only need to change the pref in `assets/panel/prefs.js`; I think the other prefs file only controls the development environment (launchpad) preference value.  Also, please reset the `assets/module-manifest.json` file.  I think we can merge after that 👍
